clientv3/client.go

```diff
@@ -86,6 +86,7 @@ func NewFromConfigFile(path string) (*Client, error) {
 // Close shuts down the client's etcd connections.
 func (c *Client) Close() error {
 	c.cancel()
+	c.Watcher.Close()
```

Comment:
Yes, but it's indirect. If there's a thousand running watches, the watch client shutdown path won't complete by the time of the goroutine leak check, so it looks like there are leaks. Directly closing the watch client ensures the client close will wait for the teardown to complete.